warehouse/templates/base.html

```diff
@@ -190,6 +190,11 @@
         <span class="notification-bar__message">{% trans %}Some features may not work without JavaScript. Please try enabling it if you encounter problems.{% endtrans %}</span>
       </div>
       </noscript>
+
+      {% block banner_messages %}
```

Comment:
Hm ? Why do we include banners via JS ? It's the same banners for everyone so they can be cached like the rest..., no ?

I imagined it more like a function `live_banners()` injected in the jinja environment, and then the snippet from warehouse/templates/includes/banner-messages.html directly in there (or via an include) ?

Maybe I'm missing something here...

Reply:
using a client side include allows banners to change without having to purge or wait for cache to clear.

Reply:
Yep, the cache was a thing here. And I also though that parallel DB queries would be better than adding an extra db query to every request pyramid gets. I'm marking this as resolved, but feel free to re-open if needed.

Reply:
Hm, it definitely won't be the hill I'll die on, but I'm not sure I understand the 2 answers. At this point, please consider I'm asking just for clarification, I'm ok with the implementation as-is and not requesting a change.

Question 1: the banner html fragment will be cached too, no ? If it's not, it's going to be a LOT of identical requests, and if it's cached, then we'll need a purge/cache timeout, no ? Maybe I missed a part ? Maybe a cache purge for a small html fragment is less costly? As it is today, the banner is baked in the page so we have the same cache as the rest of the page.

Question 2: regarding the DB justification: for me it's about having a web request with n+1 DB queries versus a web request with n queries and a web request with 1 query. As far as I can tell, 1 web request is much more expensive than 1 DB query in an already existing web request, no?

Again, not requesting change and this doesn't need to block merging the PR or anything.